### PR TITLE
Update Thompson MP from gsl/develop, minor CCPP updates, minor regression test script updates

### LINKED ISSUE #484: Error at the end of running compile.sh
## Description

Recent updates to how to build the model introduced a bug when running `compile.sh` as before:
```
./compile.sh cheyenne.intel 'APP=ATM SUITES=FV3_GFS_2017,FV3_GFS_2017_stretched 32BIT=Y' '' NO NO 2>&1 | tee compile.log
```
compiles the model correctly, but finishes with the error
```
+ echo 'Elapsed time 741 seconds. Compiling APP=ATM SUITES=FV3_GFS_2017,FV3_GFS_2017_stretched 32BIT=Y finished'
Elapsed time 741 seconds. Compiling APP=ATM SUITES=FV3_GFS_2017,FV3_GFS_2017_stretched 32BIT=Y finished
./compile.sh: line 187: COMPILE_NR: unbound variable
```
That's because of the last line in `compile.sh`:
```
echo "Compile $COMPILE_NR elapsed time $elapsed seconds. ${MAKE_OPT}" >> ${LOG_DIR}/compile_${COMPILE_NR}_time.log
```
Once this is fixed, I get:
```
./compile.sh: line 187: LOG_DIR: unbound variable
```
Maybe it's better to do:
```
elapsed=$SECONDS
echo "Elapsed time $elapsed seconds. Compiling ${MAKE_OPT} finished"
[[ -z "${LOG_DIR}" ]] && echo "Compile $COMPILE_NR elapsed time $elapsed seconds. ${MAKE_OPT}" >> ${LOG_DIR}/compile_${COMPILE_NR}_time.log
```
Is it a good idea to put stuff that depends on `rt.sh` into `compile.sh`? I know this is the main purpose of `compile.sh`, supporting `rt.sh`, but it would be nice if it worked without `rt.sh`, too.

### LINKED ISSUE #485: Compiling UFS slow because code gets recompiled unnecessarily
## Description

Some recent change (yet to be determined) reintroduced the problem that files get recompiled unnecessarily. If I run the same compile command twice, e.g.
```
./compile.sh cheyenne.intel 'APP=ATM SUITES=FV3_GFS_2017,FV3_GFS_2017_stretched,FV3_GFS_2017_stretched_nest 32BIT=Y' '' NO NO 2>&1 | tee compile.log
```
I see that the autogenerated CCPP physics caps, the entire GFDL_atmos_cubed_sphere directory and many files in fv3atm get compiled again.

This wasn't the case until recently - running the same compile command only compiled what had changed and what depended on it.

I am not aware of any changes on the CCPP side that would cause that, and any help from our cmake pros would be appreciated.

Update: this is with hash 37f0da7b82e32a15bf072eaa158399a6b0f2722c; the following hash of the model still seems to be fine:
```
commit 3be851d55572221d81a807d1c53dc909a4866154
Author: Denise Worthen <denise.worthen@noaa.gov>
Date:   Tue Mar 16 09:11:49 2021 -0400

    Updatetemplate; Fix model slowness when using threading; Update WW3 for porting and threading issues (#383)


    * CI passed at commit 464a90e prior to review.
    * Note: Orion RTs failed because of system work.

    Co-authored-by: Jessica Meixner <jessica.meixner@noaa.gov>
    Co-authored-by: Brian Curtis <brian.curtis@noaa.gov>
```

### LINKED ISSUE #496: Move baselines and input-data directory to emc.nemspara on Gaea and update rt.sh
## Description
baselines and input directories on Gaea are currently in a directory with limited access to UFS code managers, now that all code managers have emc.nemspara access we need to move them to a directory inside emc.nempara.

## Solution
DISKNM: /lustre/f2/pdata/ncep/emc.nemspara/RT

### LINKED ISSUE #499: Update Thompson MP
## Description

The author of the Thompson microphysics scheme has worked with NOAA-GSL to update and improve the Thompson MP implementation in the UFS/CCPP. The changes were made and tested in the NOAA-GSL fork, branch gsl/develop, and need to be merged into the authoritative code base.

### PR DESCRIPTION
**Pull request by climbfuji**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [X] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Description

This PR contains a whole lot of unrelated changes from various contributors
- Updates to auto-rt in PR https://github.com/ufs-community/ufs-weather-model/pull/491 from @BrianCurtis-NOAA 
- Fixes to the regression testing scripts in branch https://github.com/DusanJovic-NOAA/ufs-weather-model/tree/rt_log_dir_fix from @DusanJovic-NOAA 
- Change baseline location on gaea to where emc.nemspara can write
- Update of submodule pointers for fv3atm, ccpp-framework and ccpp-physics, and new Thompson MP lookup tables for the changes described in the associated PRs below
- Remove tests fv3_gfs_v15p2_RRTMGP and fv3_gfs_v15p2_RRTMGP_debug from `tests/rt_gnu.conf` (same errors as with other GFS v15p2 tests previously, `compute_qs: saturation vapor pressure table overflow`)

New baselines are required, because all regression tests using Thompson MP or RUC LSM will have different answers.

New input data is required, which can be added to the existing input data directory `input-data-20210324`:
```
./FV3_input_data_gsd/*V2.dat
./FV3_regional_input_data/*V2.dat
./FV3_input_data_regional_esg/*V2.dat
```

New input data copied to:

- [X] hera.intel
- [X] hera.gnu
- [X] orion.intel
- [X] cheyenne.intel 
- [X] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3

### Issue(s) addressed

Fixes https://github.com/ufs-community/ufs-weather-model/issues/499
Fixes https://github.com/ufs-community/ufs-weather-model/issues/496
Fixes https://github.com/ufs-community/ufs-weather-model/issues/484
Fixes https://github.com/ufs-community/ufs-weather-model/issues/485
Closes #491  

## Testing

Code compiles on Hera with Intel and GNU for all CCPP suites.

Regression tests pass against the existing baselines on hera.gnu and hera.intel unless Thompson MP or RUC LSM is used, as expected.

New baselines will be created after this verification is complete.

Baselines created and results verified on:
- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3

## Dependencies

https://github.com/NCAR/ccpp-framework/pull/357
https://github.com/NCAR/ccpp-physics/pull/600
https://github.com/NOAA-EMC/fv3atm/pull/267
https://github.com/ufs-community/ufs-weather-model/pull/498

### CODE REVIEW

**Comment by BrianCurtis-NOAA** [comment]
Since my PR stuff came in. After this comment should we accept passed RT and BL commits.

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: gnu
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/603877129/20210330201919/ufs-weather-model
subprocess.Popen FAILED. Exception:<class 'Exception'>
subprocess.Popen FAILED. Exception:<class 'Exception'>
subprocess.Popen FAILED. Exception:<class 'Exception'>
subprocess.Popen FAILED. Exception:<class 'Exception'>
subprocess.Popen FAILED. Exception:<class 'Exception'>
run() FAILED. Exception:<class 'Exception'>
Please make changes and add the following label back:
hera-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: gnu
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/603877129/20210330202152/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_90604
Test 013 fv3_gfsv16_ugwpv1 FAIL
Test 014 fv3_gfsv16_ugwpv1_warmstart FAIL
Test 029 fv3_gfsv16_ugwpv1_debug FAIL
BL FAILED. Exception:<class 'Exception'>
run() FAILED. Exception:<class 'Exception'>
Please make changes and add the following label back:
hera-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: gnu
Job: BL
run() FAILED. Exception:<class 'Exception'>
Please make changes and add the following label back:
hera-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: gnu
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/603877129/20210330211811/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_191994
Test 013 fv3_gfsv16_ugwpv1 FAIL
Test 014 fv3_gfsv16_ugwpv1_warmstart FAIL
Test 029 fv3_gfsv16_ugwpv1_debug FAIL
BL FAILED. Exception:<class 'Exception'>
run() FAILED. Exception:<class 'Exception'>
Please make changes and add the following label back:
hera-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
Machine: gaea
Compiler: intel
Job: BL
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/603877129/20210331125625/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_30305
Please make changes and add the following label back:
gaea-intel-BL

**Comment by MinsukJi-NOAA** [comment]
This is the fix for failing CI test. job_timestamp.txt is only available if a job runs with a scheduler. This fix has been tested on Hera and on a local Linux machine.
```
diff --git a/tests/run_test.sh b/tests/run_test.sh
index 9c6afad5..0d6010de 100755
--- a/tests/run_test.sh
+++ b/tests/run_test.sh
@@ -165,7 +165,9 @@ fi
 
 check_results
 
-cat ${RUNDIR}/job_timestamp.txt >> ${LOG_DIR}/job_${JOB_NR}_timestamp.txt
+if [[ $SCHEDULER != 'none' ]]; then
+  cat ${RUNDIR}/job_timestamp.txt >> ${LOG_DIR}/job_${JOB_NR}_timestamp.txt
+fi
 ################################################################################
 # End test
 ################################################################################
```

**Comment by BrianCurtis-NOAA** [comment]
Machine: gaea
Compiler: intel
Job: BL
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/603877129/20210331170134/ufs-weather-model
Please make changes and add the following label back:
gaea-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
Machine: gaea
Compiler: intel
Job: BL
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/603877129/20210331171947/ufs-weather-model
Please make changes and add the following label back:
gaea-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
Machine: gaea
Compiler: intel
Job: BL
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/603877129/20210331172544/ufs-weather-model
Please make changes and add the following label back:
gaea-intel-BL

File: tests/fv3_conf/compile_bsub.IN_wcoss_dell_p3 [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #BSUB -oo out
 #BSUB -eo err
 #BSUB -J @[JBNME]
-#BSUB -W 00:45
+#BSUB -W 01:00:00
```
Comment (DeniseWorthen) [inline tests/fv3_conf/compile_bsub.IN_wcoss_dell_p3]: Using #BSUB -W 00:50 allowed both compile 16 and 17 to complete--about 2800 secs each.

Comment (climbfuji) [reply tests/fv3_conf/compile_bsub.IN_wcoss_dell_p3]: Ok, let me make this change.

**Comment by climbfuji** [comment]
All, noting that the CI tests passed after I added the fix suggested by @MinsukJi-NOAA. No need to run again at the end.

**Comment by climbfuji** [comment]
Also noting that I fixed `rt_ccpp_dev.conf` and successfully created a personal baseline and verified against it on Hera using both GNU and Intel.

[rt_ccpp_dev_hera_gnu_create.log](https://github.com/ufs-community/ufs-weather-model/files/6238930/rt_ccpp_dev_hera_gnu_create.log)
[rt_ccpp_dev_hera_gnu_verify.log](https://github.com/ufs-community/ufs-weather-model/files/6238932/rt_ccpp_dev_hera_gnu_verify.log)
[rt_ccpp_dev_hera_intel_create.log](https://github.com/ufs-community/ufs-weather-model/files/6238934/rt_ccpp_dev_hera_intel_create.log)
[rt_ccpp_dev_hera_intel_verify.log](https://github.com/ufs-community/ufs-weather-model/files/6238936/rt_ccpp_dev_hera_intel_verify.log)

**Review by DeniseWorthen — APPROVED** [review]
Approved on the basis of RTs passing.

**Comment by climbfuji** [comment]
Thanks @DeniseWorthen for running the tests on WCOSS.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by climbfuji** [comment]
@BrianCurtis-NOAA auto-bl/auto-rt on gaea worked. I also ran the tests manually against your baseline, and they also passed. That system seems to work. Do I need to pull in updates from your branch before we merge this PR?

**Review by BrianCurtis-NOAA — APPROVED** [review]
baseline/RT passed. Approved

**Review by junwang-noaa — APPROVED** [review]
The .gitmodules needs to be updated after fv3 is committed

**Comment by climbfuji** [comment]
Ready to merge!

**Pull request merged**